### PR TITLE
upgrade alb-sdk to 22-1-3

### PR DESCRIPTION
**Pull request by arihantg**

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by swathinsankaran** [comment]
@arihantg could you please add jira id to the title?

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/onsi/gomega v1.24.1
 	github.com/openshift/api v3.9.0+incompatible
 	github.com/openshift/client-go v0.0.0-20201020082437-7737f16e53fc
-	github.com/vmware/alb-sdk v0.0.0-20220405050634-1a01e2eee142
+	github.com/vmware/alb-sdk v0.0.0-20230628102904-861ce312a6bf
```
Comment (swathinsankaran) [inline go.mod:13]: this version seems to be different from what is present in Ako.

Comment (swathinsankaran) [reply go.mod:13]: got it.

**Review by swathinsankaran — APPROVED** [review]
LGTM.
we have to run sanity with this image to ensure it is not breaking anything.

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Review by swathinsankaran — CHANGES_REQUESTED** [review]
@arihantg Golangci-lint is failing. Please check.

**Comment by arihantg** [comment]
build amko

**Review by swathinsankaran — APPROVED** [review]

**Comment by swathinsankaran** [comment]
could you please attach the FT results.

**Comment by arihantg** [comment]
```
**********************************************************************************************************************************
*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : test/avitest/functional/amko/test_host_rule.py   |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestHostRule.test_reset_amko                                                     |     PASSED     |       00:00:10      |   12:09:54  |   12:10:04  |
| TestHostRule.test_create_apps                                                    |     PASSED     |       00:00:11      |   12:10:04  |   12:10:16  |
| TestHostRule.test_create_hr                                                      |     PASSED     |       00:00:20      |   12:10:16  |   12:10:37  |
| TestHostRule.test_remove_aliases                                                 |     PASSED     |       00:00:20      |   12:10:37  |   12:10:58  |
| TestHostRule.test_update_aliases                                                 |     PASSED     |       00:00:20      |   12:10:58  |   12:11:19  |
| TestHostRule.test_update_duplicate_aliases_in_cluster                            |     PASSED     |       00:00:20      |   12:11:19  |   12:11:40  |
| TestHostRule.test_update_duplicate_aliases_across_cluster                        |     PASSED     |       00:00:21      |   12:11:40  |   12:12:01  |
| TestHostRule.test_delete_duplicate_aliases_in_cluster                            |     PASSED     |       00:00:10      |   12:12:01  |   12:12:11  |
| TestHostRule.test_delete_duplicate_aliases_across_cluster                        |     PASSED     |       00:00:20      |   12:12:11  |   12:12:32  |
| TestHostRule.test_cleanup                                                        |     PASSED     |       00:00:50      |   12:12:32  |   12:13:22  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 10   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 10                                                            |    Skipped:0   | Total Time:00:03:28 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+**********************************************************************************************************************************
*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name :                                                  |                |                     |             |             |
| test/avitest/functional/amko/test_amko_gslb_config.py                            |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestGslbConfig.test_default_gslb_config                                          |     PASSED     |       00:00:26      |   10:15:14  |   10:15:40  |
| TestGslbConfig.test_add_duplicate_gslb_config                                    |     PASSED     |       00:00:10      |   10:15:40  |   10:15:50  |
| TestGslbConfig.test_wrong_secret_object                                          |     PASSED     |       00:00:00      |   10:15:50  |   10:15:51  |
| TestGslbConfig.test_log_level_accepted                                           |     PASSED     |       00:00:20      |   10:15:51  |   10:16:11  |
| TestGslbConfig.test_cleanup_config                                               |     PASSED     |       00:00:00      |   10:16:11  |   10:16:11  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |    Passed: 5   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 5                                                             |    Skipped:0   | Total Time:00:00:57 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
[2023-08-18 10:18:51] :INFO: [logger_utils timed:58] calc_exc_time: 'display_summary' - 0.01 sec
Rebot framework is generating html file...

**********************************************************************************************************************************
*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name :                                                  |                |                     |             |             |
| test/avitest/functional/amko/test_amko_pool_priority.py                          |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestAmkoPoolPriority.test_create_gdp_with_priority                               |     PASSED     |       00:00:22      |   06:04:14  |   06:04:37  |
| TestAmkoPoolPriority.test_update_gdp_with_priority_update                        |     PASSED     |       00:00:10      |   06:04:37  |   06:04:48  |
| TestAmkoPoolPriority.test_update_gdp_with_default_priority                       |     PASSED     |       00:00:00      |   06:04:48  |   06:04:48  |
| TestAmkoPoolPriority.test_update_gdp_with_empty_to_some_priority                 |     PASSED     |       00:00:10      |   06:04:48  |   06:04:59  |
| TestAmkoPoolPriority.test_update_gdp_with_traffic_split_empty_priority           |     PASSED     |       00:00:00      |   06:04:59  |   06:05:00  |
| TestAmkoPoolPriority.test_cleanup_apps                                           |     PASSED     |       00:00:00      |   06:05:00  |   06:05:00  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |    Passed: 6   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 6                                                             |    Skipped:0   | Total Time:00:00:46 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
[2023-08-18 06:07:36] :INFO: [logger_utils timed:58] calc_exc_time: 'display_summary' - 0.01 sec

**********************************************************************************************************************************
*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name :                                                  |                |                     |             |             |
| test/avitest/functional/amko/test_gslb_hostrule.py                               |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestGslbHostRule.test_reset_amko                                                 |     PASSED     |       00:00:10      |   09:08:21  |   09:08:32  |
| TestGslbHostRule.test_create_apps                                                |     PASSED     |       00:00:02      |   09:08:32  |   09:08:34  |
| TestGslbHostRule.test_gdp_with_site_persistence                                  |     PASSED     |       00:00:10      |   09:08:34  |   09:08:45  |
| TestGslbHostRule.test_gdp_with_site_persistence_with_tlsenabledflag              |     PASSED     |       00:00:13      |   09:08:45  |   09:08:59  |
| TestGslbHostRule.test_gdp_with_site_health_monitor                               |     FAILED     |       00:01:32      |   09:08:59  |   09:10:31  |
| TestGslbHostRule.test_gdp_with_health_monitor_template                           |     PASSED     |       00:00:18      |   09:10:32  |   09:10:51  |
| TestGslbHostRule.test_gdp_with_hmrefs_and_hm_template                            |     PASSED     |       00:00:00      |   09:10:51  |   09:10:51  |
| TestGslbHostRule.test_gdp_with_ttl                                               |     PASSED     |       00:00:01      |   09:10:51  |   09:10:53  |
| TestGslbHostRule.test_sync_type_change                                           |     FAILED     |       00:01:43      |   09:10:53  |   09:12:37  |
| TestGslbHostRule.test_sync_type_change_on_reboot                                 |     FAILED     |       00:01:43      |   09:12:37  |   09:14:21  |
| TestGslbHostRule.test_traffic_split                                              |     FAILED     |       00:00:00      |   09:14:21  |   09:14:22  |
| TestGslbHostRule.test_gdp_with_algorithm                                         |     FAILED     |       00:01:32      |   09:14:22  |   09:15:54  |
| TestGslbHostRule.test_gdp_with_down_response                                     |     FAILED     |       00:01:32      |   09:15:54  |   09:17:27  |
| TestGslbHostRule.test_add_gslb_hostrule_with_ttl                                 |     PASSED     |       00:00:32      |   09:17:27  |   09:18:00  |
| TestGslbHostRule.test_add_gslb_hostrule_with_algorithm                           |     PASSED     |       00:00:22      |   09:18:00  |   09:18:22  |
| TestGslbHostRule.test_add_gslb_hostrule_with_health_monitor                      |     PASSED     |       00:00:32      |   09:18:22  |   09:18:55  |
| TestGslbHostRule.test_gslb_hostrule_with_health_monitor_template                 |     PASSED     |       00:00:23      |   09:18:55  |   09:19:18  |
| TestGslbHostRule.test_gslb_hostrule_with_hmrefs_and_hm_template                  |     PASSED     |       00:00:10      |   09:19:18  |   09:19:28  |
| TestGslbHostRule.test_add_gslb_hostrule_with_site_persistence                    |     PASSED     |       00:00:21      |   09:19:28  |   09:19:50  |
| TestGslbHostRule.test_gslb_hostrule_with_down_response                           |     PASSED     |       00:01:03      |   09:19:50  |   09:20:53  |
| TestGslbHostRule.test_gslb_hostrule_with_publicIP                                |     PASSED     |       00:00:43      |   09:20:53  |   09:21:37  |
| TestGslbHostRule.test_gslb_hostrule_with_priority                                |     PASSED     |       00:00:32      |   09:21:37  |   09:22:09  |
| TestGslbHostRule.test_add_delete_add_gslbhr                                      |     PASSED     |       00:00:33      |   09:22:09  |   09:22:43  |
| TestGslbHostRule.test_invalid_hm_and_site_persistence                            |     PASSED     |       00:00:34      |   09:22:43  |   09:23:18  |
| TestGslbHostRule.test_modify_fqdn_gslbhr                                         |     PASSED     |       00:00:23      |   09:23:18  |   09:23:42  |
| TestGslbHostRule.test_reboot_amko_lastmodified                                   |     FAILED     |       00:01:01      |   09:23:42  |   09:24:44  |
| TestGslbHostRule.test_cleanup_avi_resources                                      |     PASSED     |       00:00:00      |   09:24:44  |   09:24:45  |
| TestGslbHostRule.test_cleanup_apps                                               |     PASSED     |       00:00:00      |   09:24:45  |   09:24:45  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 21   |                     |             |             |
|                                                                                  |    Failed:7    |                     |             |             |
|  Total test cases: 28                                                            |    Skipped:0   | Total Time:00:16:21 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+

**********************************************************************************************************************************
*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name : test/avitest/functional/amko/test_amko_hm.py     |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestGSHealthMonitor.test_reset_amko                                              |     PASSED     |       00:00:10      |   10:21:09  |   10:21:20  |
| TestGSHealthMonitor.test_create_apps                                             |     PASSED     |       00:00:51      |   10:21:20  |   10:22:11  |
| TestGSHealthMonitor.test_ingress_route_hm_name                                   |     PASSED     |       00:00:10      |   10:22:11  |   10:22:22  |
| TestGSHealthMonitor.test_update_ingress_route_hm_name                            |     PASSED     |       00:00:21      |   10:22:22  |   10:22:43  |
| TestGSHealthMonitor.test_update_multi_path_ingress                               |     PASSED     |       00:00:11      |   10:22:43  |   10:22:55  |
| TestGSHealthMonitor.test_gses_hm_for_lb_apps                                     |     PASSED     |       00:00:00      |   10:22:55  |   10:22:55  |
| TestGSHealthMonitor.test_gses_hm_port_add_new_minimum                            |     PASSED     |       00:00:10      |   10:22:55  |   10:23:06  |
| TestGSHealthMonitor.test_gses_hm_port_add_new_maximum                            |     PASSED     |       00:00:00      |   10:23:06  |   10:23:06  |
| TestGSHealthMonitor.test_gses_delete_mimimum_port_service                        |     PASSED     |       00:00:10      |   10:23:06  |   10:23:16  |
| TestGSHealthMonitor.test_gses_rapid_health_monitor_operations                    |     PASSED     |       00:00:10      |   10:23:16  |   10:23:27  |
| TestGSHealthMonitor.test_upgrade_amko_stale_hm_deletion                          |     PASSED     |       00:00:16      |   10:23:27  |   10:23:43  |
| TestGSHealthMonitor.test_cleanup_ingress_route                                   |     FAILED     |       00:00:40      |   10:23:43  |   10:24:24  |
| TestGSHealthMonitor.test_cleanup_svc                                             |     PASSED     |       00:00:10      |   10:24:25  |   10:24:35  |
| TestGSHealthMonitor.test_cleanup_apps                                            |     PASSED     |       00:00:50      |   10:24:35  |   10:25:25  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 13   |                     |             |             |
|                                                                                  |    Failed:1    |                     |             |             |
|  Total test cases: 14                                                            |    Skipped:0   | Total Time:00:04:16 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+

*                                          AviTest Test Cases Execution Summary Report                                           *
**********************************************************************************************************************************
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
| Absolute Path:Test case                                                          |     Results    |       Duration      |  Start Time |   End Time  |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
|      Test Suite Path and Name :                                                  |                |                     |             |             |
| test/avitest/functional/amko/test_host_rule_custom_fqdn.py                       |                |                     |             |             |
| ++++++++++++++++++++++++++++++++++++++++                                         |                |                     |             |             |
| TestHostRule.test_reset_amko                                                     |     PASSED     |       00:00:10      |   11:26:53  |   11:27:04  |
| TestHostRule.test_create_apps                                                    |     PASSED     |       00:00:01      |   11:27:04  |   11:27:05  |
| TestHostRule.test_create_hr                                                      |     PASSED     |       00:00:10      |   11:27:05  |   11:27:16  |
| TestHostRule.test_create_hr_unset_includeAliases                                 |     PASSED     |       00:00:10      |   11:27:16  |   11:27:27  |
| TestHostRule.test_remove_aliases                                                 |     PASSED     |       00:00:20      |   11:27:27  |   11:27:48  |
| TestHostRule.test_toggle_includeAliases                                          |     PASSED     |       00:00:20      |   11:27:48  |   11:28:09  |
| TestHostRule.test_update_gfqdn                                                   |     PASSED     |       00:00:20      |   11:28:09  |   11:28:30  |
| TestHostRule.test_update_gfqdn_unset_includeAliases                              |     PASSED     |       00:00:12      |   11:28:30  |   11:28:42  |
| TestHostRule.test_update_aliases                                                 |     PASSED     |       00:00:21      |   11:28:42  |   11:29:03  |
| TestHostRule.test_update_aliases_unset_includeAliases                            |     PASSED     |       00:00:10      |   11:29:03  |   11:29:14  |
| TestHostRule.test_update_duplicate_aliases_in_cluster                            |     PASSED     |       00:00:21      |   11:29:14  |   11:29:35  |
| TestHostRule.test_update_duplicate_aliases_across_cluster                        |     PASSED     |       00:00:21      |   11:29:35  |   11:29:56  |
| TestHostRule.test_delete_duplicate_aliases_in_cluster                            |     PASSED     |       00:00:11      |   11:29:56  |   11:30:08  |
| TestHostRule.test_delete_duplicate_aliases_across_cluster                        |     PASSED     |       00:00:21      |   11:30:08  |   11:30:29  |
| TestHostRule.test_cleanup                                                        |     PASSED     |       00:00:50      |   11:30:29  |   11:31:19  |
| -------------------------------------------------------------------------------- |  ------------  |     -----------     | ----------- | ----------- |
|                                                                                  |   Passed: 15   |                     |             |             |
|                                                                                  |    Failed:0    |                     |             |             |
|  Total test cases: 15                                                            |    Skipped:0   | Total Time:00:04:25 |             |             |
+----------------------------------------------------------------------------------+----------------+---------------------+-------------+-------------+
```

**Pull request merged**